### PR TITLE
docs: How to use telemetry span code

### PR DESCRIPTION
**Pull request by nkomonen-amazon**
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/telemetry/spans.ts [hunk 1]
```diff
@@ -18,6 +18,14 @@ import {
 import { getTelemetryReason, getTelemetryResult } from '../errors'
 import { entries, NumericKeys } from '../utilities/tsUtils'
 
+/**
+ *
+ *
+ * For information on how to use "span" related code see CONTRIBUTING.md
```
Comment (justinmk3) [inline src/shared/telemetry/spans.ts:24]:
```suggestion
 * For information on how to use "span" related code see docs/telemetry.md
```

**Comment by justinmk3** [comment]
This is a nice doc, should probably live in `docs/telemetry.md` though. Can link to it from the "Telemetry" section of contributing.md . We will likely have more to add in the future.

**Comment by nkomonen-amazon** [comment]
@justinmk3 updated

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -275,18 +275,7 @@ then following the log with:
...
-    [telemetry/vscodeTelemetry.json](https://github.com/aws/aws-toolkit-vscode/blob/21ca0fca26d677f105caef81de2638b2e4796804/src/shared/telemetry/vscodeTelemetry.json).
-    -   Building the project will trigger the `generateClients` build task, which generates new symbols in `shared/telemetry/telemetry`, which you can import via:
-        ```
-        import { telemetry } from '../../shared/telemetry/telemetry'
-        ```
-    -   The metrics defined in `vscodeTelemetry.json` should be upstreamed to [aws-toolkit-common](https://github.com/aws/aws-toolkit-common/blob/main/telemetry/definitions/commonDefinitions.json) after launch (at the latest).
--   Metrics are dropped (not posted to the service) if the extension is running in [CI or other
-    automation tasks](https://github.com/aws/aws-toolkit-vscode/blob/21ca0fca26d677f105caef81de2638b2e4796804/src/shared/vscode/env.ts#L71-L73).
-    -   You can always _test_ telemetry via [assertTelemetry()](https://github.com/aws/aws-toolkit-vscode/blob/21ca0fca26d677f105caef81de2638b2e4796804/src/test/testUtil.ts#L164), regardless of the current environment.
+[Click here for telemetry documentation.](./docs/telemetry.md)
```
Comment (justinmk3) [inline CONTRIBUTING.md:278]:
```suggestion
See [docs/telemetry.md](./docs/telemetry.md) for notes on emitting telemetry in this project.
```

**Pull request merged**